src/common/ServerPushSetupBanner.js

```diff
@@ -0,0 +1,88 @@
+/* @flow strict-local */
+
+import React from 'react';
+import type { Node } from 'react';
+import isAfter from 'date-fns/isAfter';
+import subWeeks from 'date-fns/subWeeks';
+
+import ZulipBanner from './ZulipBanner';
+import { useSelector, useGlobalSelector, useDispatch } from '../react-redux';
+import { getIdentity, getAccount } from '../account/accountsSelectors';
+import { getIsAdmin, getRealm, getGlobalSettings } from '../directSelectors';
+import { dismissServerPushSetupNotice } from '../account/accountActions';
+import { openLinkWithUserPreference } from '../utils/openLink';
+
+type Props = $ReadOnly<{||}>;
+
+/**
+ * A "nag banner" saying the server hasn't enabled push notifications, if so
+ *
+ * When dismissed, reappears again in two weeks.
+ */
+export default function PushNotifsSetupBanner(props: Props): Node {
+  const dispatch = useDispatch();
+  const lastDismissedServerPushSetupNotice = useSelector(
+    state => getAccount(state).lastDismissedServerPushSetupNotice,
+  );
+  const pushNotificationsEnabled = useSelector(state => getRealm(state).pushNotificationsEnabled);
+  const realm = useSelector(state => getIdentity(state).realm);
+  const isAdmin = useSelector(getIsAdmin);
+  const settings = useGlobalSelector(getGlobalSettings);
+
+  let visible = false;
+  let text = '';
+  if (pushNotificationsEnabled) {
+    // don't show
+  } else if (
+    lastDismissedServerPushSetupNotice !== null
+    && isAfter(
```

Comment:
This can be just `>=` -- comparison between Date objects behaves just as you'd hope.

(And in fact if you look at the implementation of `date-fns/isAfter`, that's just what it does, after it's taken its possibly-not-actual-Date arguments and converted them into `Date`s.)